src/Simplex/Messaging/Agent/Store.hs

```diff
@@ -94,6 +94,7 @@ data DeliveryStatus
 
 type SMPServerId = Int64
 
+-- TODO rework types - decouple Transmission types from Store? Convert on the agent instead?
 class Monad m => MonadAgentStore s m where
```

Comment:
yes, I agree, we need to extract all types (and, possibly, some functions) used across Agent, Server and Client into a separate top level file (e.g. Core - tbc), so that  Server and Agent do not depend on each other.

There is no rush doing it - I think it's best to do it in the end once we can see what depends on what and decide how we want the dependency graph to look.